src/web/Navigator.tsx

```diff
@@ -137,10 +137,23 @@ export class Navigator extends RX.Navigator<NavigatorState> {
     constructor(initialProps?: Types.NavigatorProps) {
         super(initialProps);
 
+        // handle optional initialRouteStack or initialRoute props
+        let { initialRouteStack, initialRoute } = initialProps;
+        if(!Array.isArray(initialRouteStack)) {
+            initialRouteStack = [];
+        }
+
+        if(initialRouteStack.length === 0 && initialRoute) {
+            initialRouteStack.push(initialRoute);
+        }
+        
+
         // Default navigator state
         this.state = {
             sceneConfigStack: [],
-            routeStack: [],
+            presentedIndex: 0,
```

Comment:
The initialization of presentedIndex is a good fix. Initialization of transitionFromIndex isn't necessary because it's set to undefined if not initialized, and this is a fine value (effectively the same as null).